mmv1/products/firebasehosting/CustomDomain.yaml

```diff
@@ -0,0 +1,514 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "CustomDomain"
+min_version: beta
+base_url: projects/{{project}}/sites/{{site_id}}/customDomains
+self_link: projects/{{project}}/sites/{{site_id}}/customDomains/{{custom_domain}}
+create_url: projects/{{project}}/sites/{{site_id}}/customDomains?customDomainId={{custom_domain}}
+update_verb: :PATCH
+update_mask: true
+description: |
+  Manages Custom Domains for Firebase Hosting. Custom Domains link your
+  domain names with Firebase Hosting sites, allowing Hosting to serve content
+  on those domain names.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/hosting"
+  api: "https://firebase.google.com/docs/reference/hosting/rest/v1beta1/projects.sites.customdomains"
+import_format:
+  [
+    "projects/{{project}}/sites/{{site_id}}/customDomains/{{custom_domain}}",
+    "{{project}}/{{site_id}}/{{custom_domain}}",
+    "sites/{{site_id}}/customDomains/{{custom_domain}}",
+    "{{site_id}}/{{custom_domain}}",
+  ]
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  actions: ["update", "delete"]
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  post_create: templates/terraform/post_create/firebasehosting_custom_domain_wait_dns_verification.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebasehosting_customdomain_basic"
+    min_version: "beta"
```

Comment:
Nit: I believe this is implied by the `min_version` at the resource level. Unless you expect some of these examples to stay beta when the resource is promoted to GA, you should be able to remove this line.

Reply:
Done.

Reply:
Update: I have to add this back or else I got this error:

> github.com/GoogleCloudPlatform/magic-modules/mmv1/api/product.rb:139:in `version_obj_or_closest': Could not find object for version ga and product Firebase Hosting (RuntimeError)`